### DISCUSSION TITLE
Optimizing Snap Reinstallation Prompts in Flask for Unchanged Shasum

### DISCUSSION THREAD

**Discussion by Nick-1979** [MetaMask/snaps#2015] [Ideas]
Updating a dApp can become quite inconvenient, with Flask repeatedly prompting for the reinstalling of the same snap with the identical shasum. Is there a way to prevent this and have the reinstallation popup only appear when the shasum has changed?

**Comment by FrederikBolding** [comment]
@Nick-1979 Do you mean for local snaps? These are intended to be used for development purposes only and therefore the intention is to always reinstall it regardless of whether the checksum has changed

**Reply by Nick-1979** [reply]
@FrederikBolding  Imagine you're developing a dapp, and with each change in the dapp code, Metamask Flask pops up, signaling the need to reinstall the snap. Typically, a snap requires reinstallation when its code changes, indicated by a modified shasum. Therefore, finding a streamlined way to handle this process can significantly enhance the convenience of snap dApp development.
